### DISCUSSION TITLE
Incorrect Lemmas for Spanish: spacy return femenine lemmas

### DISCUSSION THREAD

**Discussion by emartinezctech** [explosion/spaCy#12645] [Help: Coding & Implementations] [answered]
Hello, 
we've notice a strage behaviour for the Spanish lemmatizer. 
We found out that in many cases it identifies the word as the lemma even though it detects that the word has gender femenine, and in Spanish lemmas are masculine plural in general. 
We thought that maybe this error has to do with a wrong POS tag, but we found that the POS tags are correct and that the error is consistent through different spacy models. 

For example, we observe this behaviour for the word 'clienta' which lemma is 'cliente'. See our code and results below: 

**_Code and results for es_dep_news_trf:_**
`nlp = spacy.load("es_dep_news_trf")
textos = ['El cliente está seguro de lo que quiere.', 'Nuestra prioridad son nuestros CLIENTES', 'La clienta no debe tener problemas durante la operación','Trabajamos para ayudar a Nuestras Clientas']
docs=list(nlp.pipe(textos))`
**Result:**
`El|el|DET['Masc'] cliente|cliente|NOUN[] está|estar|AUX[] seguro|seguro|ADJ['Masc'] de|de|ADP[] lo|él|PRON['Masc'] que|que|PRON[] quiere|querer|VERB[] .|.|PUNCT[]
Nuestra|nuestro|DET['Fem'] prioridad|prioridad|NOUN['Fem'] son|ser|AUX[] nuestros|nuestro|DET['Masc'] CLIENTES|cliente|NOUN['Masc']
La|el|DET['Fem'] clienta|clienta|NOUN['Fem'] no|no|ADV[] debe|deber|AUX[] tener|tener|VERB[] problemas|problema|NOUN['Masc'] durante|durante|ADP[] la|el|DET['Fem'] operación|operación|NOUN['Fem']
Trabajamos|trabajar|VERB[] para|para|ADP[] ayudar|ayudar|VERB[] a|a|ADP[] Nuestras|nuestro|DET['Fem'] Clientas|Clientas|PROPN[]`

**_Code and results for es_core_news_sm:**
`nlp = spacy.load("es_core_news_sm")
docs=list(nlp.pipe(textos))
for doc in docs:
    print(' '.join([token.text+'|'+token.lemma_+'|'+token.pos_+str(token.morph.get("Gender")) for token in doc]))`
**Result:**
`El|el|DET['Masc'] cliente|cliente|NOUN[] está|estar|AUX[] seguro|seguro|ADJ['Masc'] de|de|ADP[] lo|él|PRON['Masc'] que|que|PRON[] quiere|querer|VERB[] .|.|PUNCT[]
Nuestra|nuestro|DET['Fem'] prioridad|prioridad|NOUN['Fem'] son|ser|AUX[] nuestros|nuestro|DET['Masc'] CLIENTES|client|ADJ[]
La|el|DET['Fem'] clienta|clienta|NOUN['Fem'] no|no|ADV[] debe|deber|AUX[] tener|tener|VERB[] problemas|problema|NOUN['Masc'] durante|durante|ADP[] la|el|DET['Fem'] operación|operación|NOUN['Fem']
Trabajamos|trabajar|VERB[] para|para|ADP[] ayudar|ayudar|VERB[] a|a|ADP[] Nuestras|nuestro|DET['Fem'] Clientas|Clientas|PROPN[]`

**_Code and results for es_core_news_md:**
`nlp = spacy.load("es_core_news_md")
docs=list(nlp.pipe(textos))
for doc in docs:
    print(' '.join([token.text+'|'+token.lemma_+'|'+token.pos_+str(token.morph.get("Gender")) for token in doc]))`
**Result:**
`El|el|DET['Masc'] cliente|cliente|NOUN[] está|estar|AUX[] seguro|seguro|ADJ['Masc'] de|de|ADP[] lo|él|PRON['Masc'] que|que|PRON[] quiere|querer|VERB[] .|.|PUNCT[]
Nuestra|nuestro|DET['Fem'] prioridad|prioridad|NOUN['Fem'] son|ser|AUX[] nuestros|nuestro|DET['Masc'] CLIENTES|cliente|NOUN['Masc']
La|el|DET['Fem'] clienta|clienta|NOUN['Fem'] no|no|ADV[] debe|deber|AUX[] tener|tener|VERB[] problemas|problema|NOUN['Masc'] durante|durante|ADP[] la|el|DET['Fem'] operación|operación|NOUN['Fem']
Trabajamos|trabajar|VERB[] para|para|ADP[] ayudar|ayudar|VERB[] a|a|ADP[] Nuestras|nuestro|DET['Fem'] Clientas|Clientas|PROPN[]`

**_Code and results for es_core_news_lg:**
`nlp = spacy.load("es_core_news_lg")
docs=list(nlp.pipe(textos))
for doc in docs:
    print(' '.join([token.text+'|'+token.lemma_+'|'+token.pos_+str(token.morph.get("Gender")) for token in doc]))`
**Result:**
`El|el|DET['Masc'] cliente|cliente|NOUN[] está|estar|AUX[] seguro|seguro|ADJ['Masc'] de|de|ADP[] lo|él|PRON['Masc'] que|que|PRON[] quiere|querer|VERB[] .|.|PUNCT[]
Nuestra|nuestro|DET['Fem'] prioridad|prioridad|NOUN['Fem'] son|ser|AUX[] nuestros|nuestro|DET['Masc'] CLIENTES|cliente|NOUN['Masc']
La|el|DET['Fem'] clienta|clienta|NOUN['Fem'] no|no|ADV[] debe|deber|AUX[] tener|tener|VERB[] problemas|problema|NOUN['Masc'] durante|durante|ADP[] la|el|DET['Fem'] operación|operación|NOUN['Fem']
Trabajamos|trabajar|VERB[] para|para|ADP[] ayudar|ayudar|VERB[] a|a|ADP[] Nuestras|nuestra|ADJ['Fem'] Clientas|Clientas|PROPN[]`

For us, it is weird that the model is able to recognize the gender of the word correctly but it is not able to ''formulate'' the lemma correctly. 

Is there any lemmatizing function that we are missing? 

**Environment:**

Operating System: Linux
Python Version Used: 3.11.3
spaCy Version Used: 3.5.3

**Comment by adrianeboyd** [comment, marked as answer]
Lots of decisions about lemma annotation (for example, how to lemmatize pronouns or punctuation) are task-specific or corpus-specific. (Using the masculine plural for nouns sounds unusual to me, though, is there a particular Spanish corpus or dictionary that does this?)

Looking at [`UD_Spanish-AnCora`](https://github.com/UniversalDependencies/UD_Spanish-AnCora), which we're using to evaluate the rule-based Spanish lemmatizer in the `es_*` pipelines, it looks like the feminine forms of similar words are lemmatized to the feminine singular.

If you need different lemmas, you could modify the rules+exceptions for the current rule-based lemmatizer or you could potentially use the trainable lemmatizer with training data that uses the alternate forms. The data behind the rule-based lemmatizer is available here under `es_lemma_*`:

https://github.com/explosion/spacy-lookups-data/tree/master/spacy_lookups_data/data

**Comment by emartinezctech** [comment]
Thank you very much for the rapid response!

My mistake, I was thinking on the masculine SINGULAR for the base form of the words for Spanish, not the plurar. Sorry about that. 

Thanks a lot for the information about the UD_Spanish-Ancora corpus. Indeed, the femenine nouns are lemmatized as their femenine singular form. We were not aware of that.

We'll try to use the rule-based lemmatizer. 
Thanks a lot! 

Best,
Eva
